core-client/src/lib.rs

```diff
@@ -98,6 +129,67 @@ impl Future for RpcFuture {
 	}
 }
 
+/// The stream returned by a subscribe.
+pub struct SubscriptionStream {
+	recv: mpsc::Receiver<Result<Value, RpcError>>,
+}
+
+impl SubscriptionStream {
+	/// Crates a new `SubscriptionStream`.
+	pub fn new(recv: mpsc::Receiver<Result<Value, RpcError>>) -> Self {
+		SubscriptionStream { recv }
+	}
+}
+
+impl Stream for SubscriptionStream {
+	type Item = Value;
+	type Error = RpcError;
+
+	fn poll(&mut self) -> Result<Async<Option<Self::Item>>, Self::Error> {
+		match self.recv.poll() {
+			Ok(Async::Ready(Some(Ok(value)))) => Ok(Async::Ready(Some(value))),
+			Ok(Async::Ready(Some(Err(error)))) => Err(error),
```

Comment:
Good point, I've mistakenly though that the error terminates the stream. I think it's because that's what happens if you ever use `for_each` combinator:
https://docs.rs/futures/0.1.27/futures/stream/trait.Stream.html#method.for_each:

> Any error on the stream or in the closure will cause iteration to be halted immediately and the future will resolve to that error.

Which makes it even more confusing :) So disregard the comment in this implementation then, callers of `for_each` should be aware of the behaviour - the implementation is fine then.